### DISCUSSION TITLE
LUA & fluent-bit

### DISCUSSION THREAD

**Discussion by BlaBlaGITTE** [fluent/fluent-bit#7498] [Q&A] [answered]
Hi all, I'm new with LUA & fluent-bit, I've a simple question about a script:

```
function replace_message(tag,timestamp,record)
    if message == "Giving 0 client threads a chance to die gracefully"
        then
        message = "Giving 0 client threads"
     return 2, timestamp, record
    end
end
```

I want replace, the content of message field, with: "Giving 0 client threads" string, but doesn't work for me. Any help/idea?

**Comment by patrick-stephens** [comment, marked as answer]
You probably want to read up on how LUA handles tables/variables.

Fluent Bit assumes structured key-value pairs for the record, e.g. `log=....`, so it depends what keys you have in your record - this is affected by what inputs and filters you are using before the LUA filter.

I would experiment in https://calyptia.github.io/sandbox/

**Reply by BlaBlaGITTE** [reply]
thanks
